app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkView.kt

```diff
@@ -139,6 +140,10 @@ class BookmarkView(
         }
 
         bookmarkAdapter.updateData(state.tree, mode)
+        if (state.mode != oldMode) {
+            bookmarkAdapter.notifyDataSetChanged()
```

Comment:
This looks okay with how the adapter works now, but I think when #13287 lands it might be better to return false in `BookmarkAdapter.BookmarkDiffUtil.areContentsTheSame` when the mode changes. That way (I think) we'll be able to take advantage of payloads to rebind things more efficiently. We can't do that now since our current DiffUtil implementation doesn't support payloads so we'll get a noticable flicker on the change animation.

Reply:
Also if that PR lands first there will likely be merge conflicts. Sorry :(